### PR TITLE
[batch] Aggregated billing project resources infrastructure

### PR DESCRIPTION
**Pull request by jigold**
Stacked on #9346 

Changes:
- infrastructure needed for kill switch
- UI page
- Default value for the limit is None

Testing:
- In the database migration, there's two updates that populate the initial state of the aggregated_billing_resources_table. I tested this by hand using a database that hadn't been migrated previously, but this might be good to double check.
- I ran the `check_resource_aggregation` loop while running `test_batch` and made sure there were no errors.
- I tested the UI page editing the limits with negative values and gibberish by hand to make sure those failed. I also refreshed the page to make sure the values were in the database and the update worked.

So here's a PR where I convinced myself it was correct a couple of days ago, but the longer this sits, the less confident I'm going to be that there's not a mistake somewhere, especially if there are a lot of changes that need to be made to the code.

### CODE REVIEW

File: batch/sql/estimated-current.txt
Comment (danking) [inline batch/sql/estimated-current.txt:209]: CASCADE

File: batch/sql/add-aggregated-billing-project-resources.sql
Comment (danking) [inline batch/sql/add-aggregated-billing-project-resources.sql:1]: I don't see any joins on batches that use this index, am I missing something?

File: batch/sql/create-billing-projects-test-limits.sql
Comment (danking) [inline batch/sql/create-billing-projects-test-limits.sql:11]: These should go in estimated current right?

**Review by danking — DISMISSED** [review]
We can write some automated tests for this, right? We'd need the batch client to expose things like billing limits. Then we could have a test that creates two batches, gets their costs, and then the cost of their billing project and asserts that the billing project's cost is the sum of those batches.

File: batch/sql/add-aggregated-billing-project-resources.sql
Comment (jigold) [reply batch/sql/add-aggregated-billing-project-resources.sql:1]: It's for the kill switch. See part 2.

**Comment by jigold** [comment]
Stacked on #9375 as well.

**Review by danking — CHANGES_REQUESTED** [review]
Some questions

**Comment by danking** [comment]
Re-assigning to cotton for next week while I am away.

File: batch/batch/batch.py
Comment (jigold) [reply batch/batch/batch.py:232]: For the tests, I needed the float rather than the string.

File: batch/batch/front_end/front_end.py
Comment (jigold) [reply batch/batch/front_end/front_end.py:1188]: Not sure where you mean it's duplicated. I removed the other function that was doing the exact same thing now.

**Comment by jigold** [comment]
Let's get a final approval on the design doc before merging this.

File: batch/batch/front_end/front_end.py
Comment (danking) [inline batch/batch/front_end/front_end.py:1230]: This will also hide any database error as an invalid value message. I prefer:
```
try:
    ...
except Exception:
    set_message(session, f'Invalid ...')
else:
    await db.execute_update(...)
    set_message(session, f'Modified ...')
```
Unless there's a database error that we expect to get and want to treat that as an invalid value?

File: batch/batch/front_end/templates/billing_limits.html
Comment (danking) [inline batch/batch/front_end/templates/billing_limits.html:17]: I'm pretty sure this is not valid HTML. Forms shouldn't be children of tables. The fix seems easy here: move the form and the hidden input all inside the `td` that contains the limit

Comment (danking) [reply batch/batch/front_end/templates/billing_limits.html:17]: Yeah, I tried this:
```
<!DOCTYPE html>
<html lang="en">
<head><title>test</title></head>
<body>
<table>
<tbody>
<form>
<input type="hidden" name="_csrf" value="123">
<tr>
  <td>
    <input type="text" required name="limit" value="1.0">
  </td>
</tr>
<tr>
  <td>
    <input type="text" required name="limit" value="2.0">
  </td>
</tr>
</input>
</form>
</tbody>
</table>
</body>
</html>
```

in the HTML validator: https://validator.w3.org/nu/#textarea

Comment (danking) [reply batch/batch/front_end/templates/billing_limits.html:17]: This validated:

```
<!DOCTYPE html>
<html lang="en">
<head><title>test</title></head>
<body>
<table>
<tbody>
<tr>
  <td>
    <form>
    <input type="hidden" name="_csrf" value="123">
    <input type="text" required name="limit" value="1.0">
    </form>
  </td>
</tr>
<tr>
  <td>
    <form>
    <input type="hidden" name="_csrf" value="123">
    <input type="text" required name="limit" value="2.0">
    </form>
  </td>
</tr>
</tbody>
</table>
</body>
</html>
```

File: web_common/web_common/templates/header.html
Comment (danking) [inline web_common/web_common/templates/header.html:26]: I think this endif is matched to the is_developer one, so we should align it the same

File: web_common/web_common/templates/header.html
Comment (danking) [inline web_common/web_common/templates/header.html:31]: I'm not sure why this was two spaces further in above. All the anchors should be at the same level.

**Review by danking — CHANGES_REQUESTED** [review]
Just a few more cosmetic things to clean up!

**Comment by jigold** [comment]
I need to rethink the test-tiny-limit and test-zero-limit. This is going to fail every time after the first merge on the deployment.

**Comment by jigold** [comment]
Ok. The limits for the tests will need to be fixed with a REST API for editing a billing limit and a new build step that is `setup_test_batch`. I'll work on that now. But I think this can go in while I'm working on that, but #9355 should depend on the new PR.

**Comment by jigold** [comment]
Nevermind. Added tests for editing the limits to this PR. Will figure out the CLI after @catoverdrive has had a chance to do what they wanted to do with that for billing projects.

**Comment by jigold** [comment]
There's no way to fix this problem other than not run those particular tests on the production server. I think that's okay.

**Comment by jigold** [comment]
New plan. @catoverdrive already figured out how to have multiple users with different clients to be able to test the dev only endpoints such as editing a billing limit. I'm going to wait for their PR to go in and then reevaluate (#9553).

**Review by danking — DISMISSED** [review]
moving off my scorecard; dismiss when you're ready

**Comment by jigold** [comment]
Closing while I dev deploy.

**Review by danking — APPROVED** [review]

**Pull request merged**
